File: ASSIGNMENT3/EmployeeManagementSystem/src/PayrollEmployee.java
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class PayrollEmployee extends Employee {

    private List<String> certifications = new ArrayList<>();

    public PayrollEmployee(String firstName, String lastName, Date dateOfBirth,
            String employeeID, Position position, Manager manager,
            EmploymentType emplType) {
        super(firstName, lastName, dateOfBirth, employeeID, position, manager, emplType);
    }

    public void processPayroll() {
        System.out.println("Processing payroll for all employees.");
    }

    public int getEmployeesManaged() {
        if (getManager() != null && getManager().getTeamMembers() != null) {
            return getManager().getTeamMembers().size();
        }
        return 0;
    }

    public void addCertification(String cert) {
        if (cert != null && !cert.isEmpty()) {
            certifications.add(cert);
        }
    }

    public List<String> getCertifications() {
        return new ArrayList<>(certifications);
    }
}

File: Advanced_Prog_Assignment3-csharp/EmployeeManagementSystem/Models/PayrollEmployee.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using EmployeeManagementSystemCSharp.Exceptions;
using EmployeeManagementSystemCSharp.EmploymentTypes;

namespace EmployeeManagementSystemCSharp.Models
{
    public class PayrollEmployee : Employee
    {
        private readonly List<string> _certifications = new List<string>();

        public PayrollEmployee(string firstName, string lastName, DateTime dateOfBirth,
                               string employeeID, Position position, Manager manager,
                               EmploymentType emplType)
            : base(firstName, lastName, dateOfBirth, employeeID, position, manager, emplType)
        {
        }

        public void processPayroll()
        {
            System.Console.WriteLine(
                $"Payroll employee {getFirstName()} {getLastName()} is processing payroll.");
        }

        public void addCertification(string cert)
        {
            try
            {
                if (string.IsNullOrWhiteSpace(cert))
                {
                    throw new InvalidExceptions("Certification cannot be empty");
                }
                _certifications.Add(cert);
            }
            catch (InvalidExceptions e)
            {
                System.Console.Error.WriteLine("Error: " + e.Message);
            }
        }

        public List<string> getCertifications()
        {
            return new List<string>(_certifications);
        }

        public override void reportToManager(string message)
        {
            System.Console.WriteLine(
                $"Payroll employee {getFirstName()} {getLastName()} reports to manager: {message}");
        }
    }
}
